app/screens/home/channel_list/servers/index.tsx

```diff
@@ -0,0 +1,162 @@
+import type {Subscription} from 'rxjs';
+
+type UnreadMessages = {
+    mentions: number;
+    unread: boolean;
+};
+
+type UnreadSubscription = UnreadMessages & {
+    subscription?: Subscription;
+}
+
+const {SERVERS} = MM_TABLES.APP;
+const {CHANNEL, MY_CHANNEL} = MM_TABLES.SERVER;
+const subscriptions: Map<string, UnreadSubscription> = new Map();
+
+const styles = StyleSheet.create({
+    icon: {
+        alignItems: 'center',
+        justifyContent: 'center',
+        position: 'absolute',
+        zIndex: 10,
+        top: 10,
+        left: 16,
+        width: 40,
+        height: 40,
+    },
+});
+
+export default function Servers() {
+    const db = DatabaseManager.appDatabase?.database;
+    const intl = useIntl();
+    const [total, setTotal] = useState<UnreadMessages>({mentions: 0, unread: false});
+    const registeredServers = useRef<ServersModel[]|undefined>();
+    const currentServerUrl = useServerUrl();
+    const isTablet = useIsTablet();
+    const theme = useTheme();
+
+    const updateTotal = () => {
+        let unread = false;
+        let mentions = 0;
+        subscriptions.forEach((value) => {
+            unread = unread || value.unread;
+            mentions += value.mentions;
+        });
+        setTotal({mentions, unread});
+    };
+
+    const unreadsSubscription = (serverUrl: string, myChannels: MyChannelModel[]) => {
+        const unreads = subscriptions.get(serverUrl);
+        if (unreads) {
+            let mentions = 0;
+            let unread = false;
+            myChannels.forEach((myChannel) => {
+                mentions += myChannel.mentionsCount;
+                unread = unread || myChannel.isUnread;
+            });
+
+            unreads.mentions = mentions;
+            unreads.unread = unread;
+            subscriptions.set(serverUrl, unreads);
+            updateTotal();
+        }
+    };
+
+    const serversObserver = async (servers: ServersModel[]) => {
+        registeredServers.current = servers;
+        servers.forEach((server) => {
+            const {lastActiveAt, url} = server;
+            if (lastActiveAt && url !== currentServerUrl) {
+                const sdb = DatabaseManager.serverDatabases[url];
+                if (sdb?.database) {
+                    if (!subscriptions.has(url)) {
+                        const unreads: UnreadSubscription = {
+                            mentions: 0,
+                            unread: false,
+                        };
+                        subscriptions.set(url, unreads);
+                        unreads.subscription = sdb.database.
+                            get(MY_CHANNEL).
+                            query(Q.on(CHANNEL, Q.where('delete_at', Q.eq(0)))).
+                            observeWithColumns(['mentions_count', 'has_unreads']).
+                            subscribe(unreadsSubscription.bind(undefined, url));
+                    }
+                }
```

Comment:
Should we abstract the database operations (in this case subscriptions) from the component?
In other words, should we move this piece of code to maybe a `actions->subscriptions` or `actions->local->subscriptions` folder?

That way we would have here only `subscribeUnreads(unreadSubscription)`.

Reply:
I'll give it a try ;)